rfc/20241023-rfc-tracking-issues.md

```diff
@@ -0,0 +1,209 @@
+
+This proposal _does not_ significantly change the earlier stage of the authoring and approval of an RFC, except for making it explicit that it's okay for an RFC to be approved and merged even when there is no immediate plan to implement it, since we'll now have a new way to represent that situation.
+
+This proposal is also primarily discussing the process for _feature implementation_ RFCs, and not for other kinds of RFCs such as development process RFCs. Non-feature RFCs are currently far less common and so we don't have as much experience with them and thus it would be premature to try to make their process significantly more prescriptive.
+
+(The ideas in this proposal are _broadly_ based on some similar ideas from the Rust project's RFC process, although the proposed details are simpler because the OpenTofu project's current scale is considerably smaller than Rust's.)
+
+## Proposed Solution
+
+As noted above, this proposal effectively expands on what is currently the last step in the existing RFC process document:
+
+> 6. The RFC is Merged, and the Core Team creates issues in the relevant repositories to track the work required to implement the RFC.
+
+Under this proposal, immediately after merging an accepted RFC the core team creates a single "tracking issue" that represents the work to implement that RFC. The tracking RFC title is always the prefix "RFC tracker: " followed by the same title used in the RFC document itself. For now, tracking issues belong in the main `opentofu/opentofu` repository regardless of which codebases the RFC proposes to modify, though [we might change that in future](#moving-rfcs-and-their-tracking-issues-into-a-separate-repository).
+
+The tracking issue description includes the following information:
+
+- A link to the RFC text itself, and the PR(s) where the document was previously discussed.
+- Links to each of the feature request issues that were identified at the beginning of the RFC text, just to make backlinks appear on those other issues linking to the tracking issue.
+- A checkbox list of blockers that must be cleared before implementation could begin, or at least before any implementation could be merged. If the blockers are work described by other RFCs, these items link directly to each other RFC's tracking issue.
+
+    (It may be appropriate to begin a partial implementation that then remains unmerged until blockers are cleared, but the author must understand that their initial implementation may need significant revisions or replacement once the blockers have been resolved.)
+- A checkbox list summarizing the work that must be completed before the RFC is considered fully implemented. This includes both the direct development work and other related tasks such as updating the documentation and coordination with other projects that might be run concurrently.
```

Comment:
Since my scope here was focused mainly on the tracking issues rather than the RFC content I described non-coding tasks as something to be listed _in the tracking RFC_, but I think you're right that it could be helpful for the RFC to include the initial list of non-coding tasks (and possibly also coding tasks?) that would get copied directly into the tracking issue when it's first created.

Does that match what you were imagining, or did I misunderstand the idea? :thinking: